jelixir/lib/jelixir.ex

```diff
@@ -20,6 +24,74 @@ defmodule JelixirLib do
     end
   end
 
+  def conver(task, file_name_string) when task == :schema do
+    with {:read_file_result, {:ok, file_content}} <-
+           {:read_file_result, read_file(file_name_string, ".ex")} do
+      if !File.exists?(@default_folder_out) do
+        File.mkdir(@default_folder_out)
+      end
+
+      parse_schema_result =
+        String.split(file_content, "\n")
+        |> Enum.map(fn line ->
+          line |> String.trim() |> read_line_schema()
+        end)
+        |> Enum.filter(fn item -> item != {"skip", nil} end)
+        |> Enum.into(%{})
+
+      module_name = parse_schema_result["module_name"]
+      schema_name = parse_schema_result["schema_name"]
+
+      node_result = Map.drop(parse_schema_result, ["module_name", "schema_name"])
+      create_gen(module_name, schema_name, node_result)
+    else
+      {:read_file_result, _} -> IO.inspect("Can't read file with name: #{file_name_string}")
+      {:name_result, _} -> IO.inspect("Can't parse file with name: #{file_name_string}")
+    end
+  end
+
+  def conver(_task, _file_name_string) do
+    {:error, "No thing to do"}
+  end
+
+  def read_line_schema(module_name_result, nil, nil) do
+    with false <- is_nil(module_name_result),
+         [_line, module_name] <- module_name_result do
+      {"module_name", module_name}
+    else
+      _ -> {"skip", nil}
+    end
+  end
+
+  def read_line_schema(nil, schema_name_result, nil) do
+    with false <- is_nil(schema_name_result),
+         [_line, schema_name] <- schema_name_result do
+      {"schema_name", schema_name}
+    else
+      _ -> {"skip", nil}
+    end
+  end
+
+  def read_line_schema(nil, nil, pattern_field_result) do
+    with false <- is_nil(pattern_field_result),
+         [_line, field_name, field_type] <- pattern_field_result do
+      {field_name, ":#{field_type}"}
+    else
+      _ -> {"skip", nil}
+    end
+  end
+
+  def read_line_schema(_module_name_result, _schema_name_result, _pattern_field_result) do
+    {"skip", nil}
+  end
+
+  def read_line_schema(flat_line) do
+    module_name_result = Regex.run(@regex_pattern_module_name, flat_line)
+    schema_name_result = Regex.run(@regex_pattern_schema_name, flat_line)
+    pattern_field_result = Regex.run(@regex_pattern_field, flat_line)
```

Comment:
Hello @ttpho, I would like to suggest the implementation of read_line_schema function

```
  def read_line_schema(flat_line) do
    module_name_result = Regex.run(@regex_pattern_module_name, flat_line)
    schema_name_result = Regex.run(@regex_pattern_schema_name, flat_line)
    pattern_field_result = Regex.run(@regex_pattern_field, flat_line)

    matched_result = 
      [module_name_result, schema_name_result, pattern_field_result]
      |> Enum.filter(&(!is_nil(&1)))
    
    case matched_result do
      [_line, module_name] ->
        {:module_name, module_name}

      [_line, schema_name] ->
        {:schema_name, schema_name}

      [_line, field_name, field_type] ->
        {field_name, field_type}

      _ ->
        {:skip, nil}
    end
  end
```

I think it will be easier to understand. Hope you consider to update this function :)

Reply:
@quangvo09 Thank you for your suggestion. But I don't think it is a good idea.

Look the condition, because `module_name` and `schema_name` is a `String`,  so this case `matched_result` can' math this result.
```
    [_line, module_name] ->
        {:module_name, module_name}

      [_line, schema_name] ->
        {:schema_name, schema_name}
```

Reply:
@ttpho yes I got it

Reply:
Hello @ttpho, I would like to give you another version of `read_line_schema` function

Update regex pattern
```
@regex_pattern_module_name ~r/(defmodule)\s+([0-9a-z_-]+)\s+do/im
@regex_pattern_schema_name ~r/(schema)\s+"([0-9a-z_-]+)"\s+do/im
@regex_pattern_field ~r/(field)\s{0,}\(?\s{0,}:([0-9a-z_-]+),\s{0,}:([0-9a-z_-]+)\)?/im
```

The implementation of `read_line_schema ` function
```
 def read_line_schema(flat_line) do
    module_name_result = Regex.run(@regex_pattern_module_name, flat_line)
    schema_name_result = Regex.run(@regex_pattern_schema_name, flat_line)
    pattern_field_result = Regex.run(@regex_pattern_field, flat_line)

    matched_result = 
      [module_name_result, schema_name_result, pattern_field_result]
      |> Enum.find(&(!is_nil(&1)))
    
    case matched_result do
      [_line, "defmodule", module_name] ->
        {:module_name, module_name}

      [_line, "schema", schema_name] ->
        {:schema_name, schema_name}

      [_line, "field", field_name, field_type] ->
        {field_name, field_type}

      _ ->
        {:skip, nil}
    end
  end
```